lib/uiautomator.js

```diff
@@ -1,225 +1,108 @@
-"use strict";
-
-var _ = require('underscore'),
-    net = require('net'),
-    logger = require('./logger.js').get();
-
-var noop = function () {};
-
-var UNKNOWN_ERROR = {
-  status: 13,
-  value: "Tried to send command to non-existent Android socket, " +
-    "maybe it's shutting down?"
-};
-
-var UiAutomator = function (adb, opts) {
-  this.adb = adb;
-  this.proc = null;
-  this.cmdCb = null;
-  this.socketClient = null;
-  this.restartBootstrap = false;
-  this.onSocketReady = noop;
-  this.alreadyExited = false;
-  this.onExit = noop;
-  this.shuttingDown = false;
-  this.webSocket = opts.webSocket;
-  this.systemPort = opts.systemPort;
-  this.resendLastCommand = function () {};
-  this.logNoColors = opts.logNoColors;
-};
-
-UiAutomator.prototype.start = function (readyCb) {
-  this.adb.killProcessesByName('uiautomator', function (err) {
-    if (err) return readyCb(err);
-    logger.info("Running bootstrap");
-    var args = ["shell", "uiautomator", "runtest", "AppiumBootstrap.jar", "-c",
-        "io.appium.android.bootstrap.Bootstrap"];
-
-    this.alreadyExited = false;
-    this.onSocketReady = readyCb;
-
-    this.proc = this.adb.spawn(args);
-    this.proc.on("error", function (err) {
-      logger.error("Unable to spawn adb: " + err.message);
-      if (!this.alreadyExited) {
-        this.alreadyExited = true;
-        readyCb(new Error("Unable to start Android Debug Bridge: " +
-          err.message));
-      }
-    }.bind(this));
-    this.proc.stdout.on('data', this.outputStreamHandler.bind(this));
-    this.proc.stderr.on('data', this.errorStreamHandler.bind(this));
-    this.proc.on('exit', this.exitHandler.bind(this));
-  }.bind(this));
-};
-
-UiAutomator.prototype.setExitHandler = function (onExit) {
-  this.onExit = onExit;
-};
-
-UiAutomator.prototype.exitHandler = function (code) {
-  this.cmdCb = null;
-  if (this.socketClient) {
-    this.socketClient.end();
-    this.socketClient.destroy();
-    this.socketClient = null;
+import events from 'events';
+import { SubProcess } from 'teen_process';
+import log from './logger';
+
+class UiAutomator extends events.EventEmitter {
+  constructor (adb) {
```

Comment:
i think we want some validation that adb exists (throw an error if you construct this without an adb object for example)